pkg/virt-controller/watch/snapshot/restore.go

```diff
@@ -565,9 +569,46 @@ func (t *vmRestoreTarget) Reconcile() (bool, error) {
 	}
 	t.vm = newVM
 
+	for _, dvt := range newTemplates {
```

Comment:
I think this should be out of the Reconcile function, instead add a reconcileDatavolumes after the call the the Reconcile, also it should be done similar to the reconcilerestorevolumes in the way that you need to wait until the function will return that it didnt create any new data volume and didnt need to wait anymore- i.e that the data volume was created successfully (should be pretty fast since it is populated).. in the way that its written now you create the datavolume but dont wait for them to succeed which might cause a problem if they will never be succeeded..

Reply:
after you do that the ifs in the updateVMRestore functions will be even worst then now, and I think probably this whole function should be changed to be with different phases and be a switch case but it will be a part of a different PR :)

Reply:
@ShellyKa13 added reconcileDataVolumes, and on also flattened updateVMRestore.

Reply:
I'm not sure `reconcileDataVolumes` should be in the `restoreTarget` interface.  Would it still make sense for a `VirtualMachineInstance` target?  They don't have DataVolumeTemplates.

Reply:
Not sure I get you. How would you create the DVs without the DataVolumeTemplates?

Reply:
DataVolumes of course still have to get created for VM targets but I think it should happen in `vmRestoreTarget.Reconcile()`.

Reply:
I still don't understand you....
Do you suggest dropping the new `ReconcileDataVolumes()` and moving its logic to `vmRestoreTarget.Reconcile()`?

Even before this PR, the DV creation in `VMController.handleDataVolumes()` was based on the `vm.Spec.DataVolumeTemplates`.